SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -0,0 +1,906 @@
++    if (!pbOutFile)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_IO;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    pData = (TDNFFilterData *) malloc(sizeof(TDNFFilterData));
++    if (!pData)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    pData->nBufferMaxLen = 5000;
++    pData->pszElementBuffer = (char *)malloc(pData->nBufferMaxLen * sizeof(char));
++    if (!pData->pszElementBuffer)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_MEMORY;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++    bzero(pData->pszElementBuffer, pData->nBufferMaxLen);
++    pData->pbOutfile = pbOutFile;
++    pData->nSearchTime = nEpochTime;
++    pData->nDepth = 0;
++    pData->nBufferLen = 0;
++    pData->nInPackage = 0;
++    pData->nPrintPackage = 0;
++    pData->nTimeFound = 0;
++
++    //create parser
++    bParser = XML_ParserCreate(NULL);
++    if (!bParser)
++    {
++        dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++        BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++    }
++
++    XML_SetUserData(bParser, pData);
++    XML_SetElementHandler(bParser, TDNFFilterStartElement, TDNFFilterEndElement);
++    XML_SetCharacterDataHandler(bParser, TDNFFilterCharDataHandler);
++
++    //parse XML
++    fprintf(pbOutFile, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n");
++    int nInputEof;
++    do
++    {
++        void * pszXMLParseBuffer = XML_GetBuffer(bParser, BUFSIZ);
++        if (!pszXMLParseBuffer)
++        {
++            fprintf(stderr, "Couldn't allocate memory for buffer\n");
++            dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++            BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++        }
++
++        const size_t len = fread(pszXMLParseBuffer, 1, BUFSIZ - 1, pbInFile);
++        ((char *)pszXMLParseBuffer)[len] = '\0';
++
++        nInputEof = feof(pbInFile);
++
++        if (XML_ParseBuffer(bParser, (int)len, nInputEof) == XML_STATUS_ERROR)
++        {
++            fprintf(stderr,
++                "Parse error at line %lu:\n%s\n",
++                XML_GetCurrentLineNumber(bParser),
++                XML_ErrorString(XML_GetErrorCode(bParser)));
++            dwError = ERROR_TDNF_TIME_FILTER_PARSE;
++            BAIL_ON_TDNF_TIME_FILTER_ERROR(dwError);
++        }
++    } while (!nInputEof);
++
++cleanup:
++    if (pData)
++    {
++        if (pData->pszElementBuffer) {
++            free(pData->pszElementBuffer);
++        }
++        free(pData);
++    }
++
++    if (bParser)
++    {
++        XML_ParserFree(bParser);
++    }
++
++    if (pbOutFile)
++    {
++        fclose(pbOutFile);
++    }
++
++    if (pbInFile)
++    {
++        fclose(pbInFile);
++    }
++
++    return pszOutFilePath;
++error:
++    goto cleanup;
++}
++// #### END XML FILTER CODE ####
++
+ uint32_t
+ SolvLoadRepomd(
+     Repo* pRepo,
+@@ -53,6 +569,7 @@ SolvLoadRepomdPrimary(
+ {
+     uint32_t dwError = 0;
+     FILE *fp = NULL;
++
```

Comment:
looks like only newlines added here and a few lines below.  We should undo that change

Reply:
removed.